src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -425,21 +421,21 @@ internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingCon
             foreach (var entry in results)
             {
                 var result = entry.Value;
-                if (result != null)
+                if (result.IsModelSet)
                 {
                     var propertyMetadata = entry.Key;
                     SetProperty(bindingContext, modelExplorer, propertyMetadata, result);
+                }
 
-                    var propertyValidationNode = result.ValidationNode;
-                    if (propertyValidationNode == null)
-                    {
-                        // Make sure that irrespective of whether the properties of the model were bound with a value,
-                        // create a validation node so that these get validated.
-                        propertyValidationNode = new ModelValidationNode(result.Key, entry.Key, result.Model);
-                    }
-
-                    validationNode.ChildNodes.Add(propertyValidationNode);
+                var propertyValidationNode = result.ValidationNode;
+                if (propertyValidationNode == null)
+                {
+                    // Make sure that irrespective of whether the properties of the model were bound with a value,
+                    // create a validation node so that these get validated.
+                    propertyValidationNode = new ModelValidationNode(result.Key, entry.Key, result.Model);
                 }
+
+                validationNode.ChildNodes.Add(propertyValidationNode);
```

Comment:
This block has different semantics than it did. Did not previously add `ModelValidationNode`s when no data existed at all (despite the comment just above). Suspect `result == ModelBindingResult.NoResult` should be handled separately from `!result.IsModelSet`.